### ISSUE TITLE
Track number of downloads for free cardshop image

### ISSUE THREAD

**Issue by Popolechien** [kiwix/cardshop#229]
Just to have an idea of our impact, it would be useful to track how many people download the free 16Gb RPi image from the cardshop sales page.

**Comment by rgaudin** [comment]
Haven't we already had this discussion?

We *could* record the number of requests to `https://api.cardshop.hotspot.kiwix.org/auto-images/demo/redirect/http` and `https://api.cardshop.hotspot.kiwix.org/auto-images/demo/redirect/torrent` but I don't see the value: 

- you already have stats for that shop page on matomo
- requests to those URLs are hardly convertible to number of downloads. Especially given it's a 16GB file and we don't advertise its size on the page so most curious visitor of that page would click.

@kelson42 what do you think? Is it valuable enough to record? Should it be viewable from the UI or should we just keep it in DB to look at it from time to time?

**Comment by kelson42** [comment]
I think the request of @Popolechien is legitim and might be implemented. I'm would prefer that all go through a redirection HTML page (itself calling matomo).

**Comment by rgaudin** [comment]
OK ; you didn't answer the UI question.

I think the html redirect is poor UX but I understand the why.
- What would be displayed on that page (that page will stay up after redirect as the content-type would trigger a download)
- How long before the redirection? Matomo script is async

If you're only interested in the number of downloads, a possibility is to trigger a matomo request from the backend code.

**Comment by kelson42** [comment]
> I think the html redirect is poor UX but I understand the why.

I agree, it is just the best compromise beetween multiple constraints I have in mind.

If we do so in Cardshop level, this will probably be needed in other use cases.

>     * What would be displayed on that page (that page will stay up after redirect as the content-type would trigger a download)

A simple message telling that a redirect will happen in 3 seconds.

>     * How long before the redirection? Matomo script is asyn

3 seconds should be enough.

**Comment by kelson42** [comment]
@rgaudin An other solution is that the the python makes the matomo API call. That ways we could do with a "transparent HTTP redirection.

**Comment by rgaudin** [comment]
> @rgaudin An other solution is that the the python makes the matomo API call. That ways we could do with a "transparent HTTP redirection.

That's exactly what I suggested… but in this case we loose the screen resolution and other metadata.

**Comment by kelson42** [comment]
> in this case we loose the screen resolution and other metadata.

This does not really matter here.
